crates/core/tedge_agent/src/file_transfer_server/http_rest.rs

```diff
@@ -54,327 +69,370 @@ impl HttpConfig {
             ..self
         }
     }
-
-    pub fn file_transfer_end_point(&self) -> String {
-        format!("{}file-transfer/*", self.file_transfer_uri)
-    }
-
-    /// Return the path of the file associated to the given `uri`
-    ///
-    /// Check that:
-    /// * the `uri` is related to the file-transfer i.e a sub-uri of `self.file_transfer_uri`
-    /// * the `path`, once normalized, is actually under `self.file_transfer_dir`
-    pub fn local_path_for_uri(&self, uri: String) -> Result<Utf8PathBuf, FileTransferError> {
-        let ref_uri = uri.clone();
-
-        // The file transfer prefix has to be removed from the uri to get the target path
-        let path = uri
-            .strip_prefix(&self.file_transfer_uri)
-            .ok_or(FileTransferError::InvalidURI { value: ref_uri })?;
-
-        // This path is relative to the file transfer dir
-        let full_path = self.data_dir.join(path);
-
-        // One must check that once normalized (i.e. any `..` removed)
-        // the path is still under the file transfer dir
-        let clean_path = clean_utf8_path(&full_path);
-        if clean_path.starts_with(&self.data_dir) {
-            Ok(clean_path)
-        } else {
-            Err(FileTransferError::InvalidURI {
-                value: clean_path.to_string(),
-            })
-        }
-    }
-}
-
-fn clean_utf8_path(path: &Utf8Path) -> Utf8PathBuf {
-    Utf8PathBuf::from(path_clean::clean(path.as_str()))
 }
 
-fn separate_path_and_file_name(input: &Utf8Path) -> Option<(Utf8PathBuf, String)> {
-    let (relative_path, file_name) = input.as_str().rsplit_once('/')?;
-
-    let relative_path = Utf8PathBuf::from(relative_path);
-    Some((relative_path, file_name.into()))
+fn separate_path_and_file_name(input: &Utf8Path) -> Option<(&Utf8Path, &str)> {
```

Comment:
The semantics of this function are completely different to what was originally written (and very directly tested (see https://github.com/thin-edge/thin-edge.io/blob/ccff7ee2720fa688e19e0cdd5b4bc7c1c3b3bb79/crates/core/tedge_agent/src/file_transfer_server/http_rest.rs#L250-L261)). However these tests were not testing anything the user could achieve, as the API did path cleanup (which removed trailing slashes) and (I think) the code we call afterwards would have returned an error if we had been able to call this with a trailing slash (as we would have created the entire path as directories and then attempted to create a file with an empty name inside that).